### ISSUE TITLE
The commands az storage container set-permission and get-permission should not support --auth-mode login

### ISSUE THREAD

**Issue by milope** [Azure/azure-cli#28597]
**Related command**
The following:

```[bash]
az storage container set-permissions --auth-mode login  etc
az storage container show-permissions --auth-mode login etc
```

**Is your feature request related to a problem? Please describe.**
The [documentation](https://learn.microsoft.com/cli/azure/storage/container?view=azure-cli-latest#az-storage-container-set-permission) states using `--auth-mode login` allowed.

Neither of these APIs allow for any authorization aside from Shared Key:

- [Get Container ACL](https://learn.microsoft.com/rest/api/storageservices/get-container-acl#authorization)
- [Set Container ACL](https://learn.microsoft.com/rest/api/storageservices/set-container-acl#authorization)

**Describe the solution you'd like**
If possible, please remark the command as not supporting `--auth-mode login`.

**Describe alternatives you've considered**
Using shared key should suffice.

**Additional context**
The documentation may be misleading.

**Comment by yonzhan** [comment]
Thank you for opening this issue, we will look into it.

**Comment by calvinhzy** [comment]
I have verified that --auth-mode login does indeed not work, will update the command to not allow this and update documentation.

**Comment by krinsman** [comment]
I'm not sure whether this is hijacking the thread, but 

1. Should the same reasoning also apply to disallowing the `--sas-token` flag for those commands (`show-permission` and `set-permission`)?
2. Should the same reasoning also affect the error messages for the `az storage container create` command when the `--public-access` flag is used with it?

**Context:** I ran into the issue where I tried to create a container with blob anonymous access, without having first allowed it for the whole storage account. The `az storage container create` command failed as a result when I tried to include the `--public-access blob` flag, and it took a long time to realize why (there were no useful error messages, even with the `--debug` flag), during which time I had tried switching the authorization method from shared key to SAS token. 

After figuring out that issue, I then tried to use the `set-permission` command on the container using a SAS token, and that failed with a misleading error message about network policy. And then I tried to debug using the `show-permission` command but had the same issue. Only after yet another long time (when I found this thread via Google search) was I able to diagnose what the issue was, and switched back to using shared key authorization (despite [the documentation saying such authorization is "legacy"](https://learn.microsoft.com/en-us/azure/storage/blobs/authorize-data-operations-cli#authorize-with-the-account-access-key)) and finally everything worked.

Basically disallowing `--sas-token` flag for `az storage container show-permission` and `az storage container set-permissions` commands would have saved me a lot of time.
